main.go

```diff
@@ -95,114 +101,171 @@ func getEtcdConfig() (etcd.Config, error) {
 	return config, nil
 }
 
-func getGlobalASN(api etcd.KeysAPI) (uint32, error) {
-	res, err := api.Get(context.Background(), fmt.Sprintf("%s/global/as_num", CALICO_BGP), nil)
+type Server struct {
+	t         tomb.Tomb
+	bgpServer *bgpserver.BgpServer
+	cli       *calicocli.Client
```

Comment:
It's a minor nit, but I find cli as a contraction for client confusing since cli is generally means command line interface.  Would be nicer to expand to client.